ambry-tools/src/main/java/com/github/ambry/cloud/azure/AzureCompactionTool.java

```diff
@@ -133,6 +133,11 @@ private PartitionPathId(String path) {
       return new byte[0];
     }
 
+    @Override
+    public long getId()  {
+      return 0;
```

Comment:
0 is a valid id. maybe you can return -1 to catch issue earlier if this class is widely used.